### DISCUSSION TITLE
The quarto for Linux tarball works on macOS, how come?

### DISCUSSION THREAD

**Discussion by b-rodrigues** [quarto-dev/quarto-cli#6641] [Q&A] [answered]
### Description

Hi all

disclaimer: I have 0 experience with macOS so maybe my question is very obvious. I am trying to package quarto for Nix for macOS. Nix is a package manager that works on Linux, Windows with WSL2 and macOS. For now, quarto on Nix is only available for Linux, and I wanted to help package the macOS version.
I was able to do so, but during the discussions with other packagers, it turns out that the tarball for Linux runs on macOS (at least for the intel architecture, but apparently it works even on Apple Sillicon architectures). Here’s the discussion: https://github.com/NixOS/nixpkgs/issues/251957#issuecomment-1697042398

So I was wondering: is this expected behaviour? What are the differences between the Linux and the macOS tarballs? I’ve also discussed with @cderv on the French R users Slack about this, and was wondering if the next release of quarto (1.4) will have an intel and apple sillicon specific tarballs?

Cheers

**Comment by cscheid** [comment, marked as answer]
(my disclaimer I don't know anything Nix)

I think you have it slightly wrong, and what's happening is that the _nix_ linux tarball works on macOS, because whoever packaged that is not shipping any of the binary dependencies. _Our_ tarballs ship binary dependencies with them, and so are (clearly) dependent on architectures.

**Comment by b-rodrigues** [comment]
Ok I think I got it. After your answer I tried the Linux tarball on my hackintosh vm (without nix), and `quarto --version` worked, which made me think that the linux tarball would always work. But then I tried `quarto check` and it failed, because deno was not able to run on that architecture. At first I thought that the nix "package" was nothing more than a script that downloads the tarball from Quarto's github repository, and then moves the bin/ and share/ folders to the right spot, and it does do that, but it also downloads a version of `deno` from its repos as well. So it does download the Linux tarball, but then also downloads the right `deno` binary for Quarto to use. So while the the Nix expression downloads the Linux tarball even on macOS, it provides `deno` from its repositories.

Thanks for helping me see!
